src/lib/FreeC/Backend/Agda/Syntax.hs

```diff
@@ -29,17 +39,21 @@ import           Agda.Syntax.Position
 
 -- | Creates a (not qualified) Agda variable name from a 'String'.
 name :: String -> Name
-name ident = Name NoRange InScope [Id ident]
+name str = Name NoRange InScope [Id str]
 
 -- | Create a qualified identifier given a local identifier as 'Name' and a
 --   list of module 'Name's.
 qname :: [Name] -> Name -> QName
-qname modules ident = foldr Qual (QName ident) modules
+qname modules n = foldr Qual (QName n) modules
 
 -- | Creates a qualified name using an empty list of module names.
 qname' :: Name -> QName
 qname' = qname []
 
+nextQName :: QName -> QName
+nextQName (Qual n qn) = Qual (nextName n) qn
```

Comment:
Not sure what happens here: in case of `Qual` the `ModName` is changed but not `qn`? Again, maybe we can rename `n` here to s.th. with more semantic meaning like `modName`.

Reply:
Its just wrong. The correct definition is of course.
```haskell
nextQName :: QName -> QName
nextQName (Qual n qn) = Qual n $ nextQName qn
nextQName (QName n  ) = QName $ nextName n
```

Reply:
That makes more sense indeed. That is, `ModName` is just a type synonym for `Name, which got us into trouble here ; )

Nevertheless, I'd like to see a better variable name than `n`.